authority/provisioner/keystore.go

```diff
@@ -133,3 +137,13 @@ func getCacheJitter(age time.Duration) time.Duration {
 func getExpirationTime(age time.Duration) time.Time {
 	return time.Now().Truncate(time.Second).Add(age)
 }
+
+// abs returns the absolute value of n using the two's complement form.
+//
+// It will overflow with math.MinInt64 in the same way a branching version
+// would, this is not a problem because maxAgeRegex will block negative numbers
+// and the logic will never produce that number.
+func abs(n time.Duration) time.Duration {
+	y := n >> 63
```

Comment:
ok, you were right. this scared me. I see you've explained what's going on but I still don't understand. Is there a link or something that you can post to a more thorough explanation?